lib/steps.js

```diff
@@ -180,66 +178,42 @@ When(/^(some|all)(?: of)?(?: the(?:se)? var(?:iable)?s are used)?/, {timeout: -1
-  // See if this is the terminating case already!
-  let { question_has_id, id, id_matches } = await scope.examinePageID( scope, scope.final_question_id );
-
   // Until we reach the final page or hit an error
+  let original_num_vars = var_data.length;  // For a check at the end
   let remaining_vars = var_data;
+  
+  let { question_has_id, id, id_matches } = await scope.examinePageID( scope, target_id );
   while ( !id_matches ) {
-
-    let promise = new Promise(function( resolve, reject ) {
+    let custom_timeout = new Promise(function( resolve, reject ) {
       // Set up the timeout
       let page_timeout = setTimeout(function() {
-        reject(`Variable table attempt did not resolve within ${ scope.timeout } ms at "${ id }". Check the error screenshot.`);
+        let timeout_message = `The target id was "${ target_id }", but it took too long` +
+        ` to try to reach it (over ${ scope.timeout/1000/60 } min). The test got stuck on "${ id }".` +
+        ` Last remaining variables detected were ${ JSON.stringify(remaining_vars) }`
+        reject( timeout_message );
       }, scope.timeout);
 
       // self-invoke an anonymous async function so we don't have to
       // abstract the details of resolving and rejecting.
       (async function() {
         try {
+          let { question_has_id, id, id_matches } = await scope.examinePageID( scope );
           // Some errors happen in there to be closer to the relevant code. We'd have to try/catch anyway.
-          let leftovers = await scope.processVar(scope, { how_many, var_data: remaining_vars, id });
+          let leftovers = await scope.processVar(scope, { var_data: remaining_vars, id });
           resolve( leftovers );
         } catch (err) {
           reject( err );
         } finally {
           clearTimeout( page_timeout );  // prevent temporary hang at end of tests
         }
       })();
-    });  // ends promise proccessVar timeout
+    });  // ends custom_timeout for proccessVar
 
-    remaining_vars = await promise;
+    remaining_vars = await custom_timeout;
     expect( Array.isArray(remaining_vars) ).to.be.true;
-
-    ({ question_has_id, id, id_matches } = await scope.examinePageID( scope, scope.final_question_id ));
+    ({ question_has_id, id, id_matches } = await scope.examinePageID( scope, target_id ));
   }  // ends while !id_matches
-
-  let remaining_json = JSON.stringify(remaining_vars);
-  let target_id = scope.final_question_id
-
-  // if the id does not match we did not reach the desired screen
-  let target_not_reached_msg = `We needed to get to the question id "${ target_id }",` +
-    ` but all the variables that can be used have been used and the question id is only "${ id }".` +
-    ` The remaining variables are ${ remaining_json }`
-  expect( id_matches, target_not_reached_msg ).to.be.true;
```

Comment:
Do we not display this error message anymore? I don't see this error message in the code anymore.

Reply:
The target id being reached is the only way the loop will stop, so if the target can't be reached, something else will have to halt the loop, such as an error. I _think_ I've managed to throw an error for all the cases that would occur, but if you see something I've missed, let me know. Remember that when a variable can't be found on the page, the code tries to hit a 'continue' type button. I _think_ these are the situations that I tried to cover, though I may be missing something that I thought of and addressed in the code:
1. The continue button is pressed, but the user can't continue - they get a validation message or a system error.
1. The code tries to hit the continue button, but it doesn't exist.
1. The code tries to set a variable, but its element can't be interacted with (it's hidden or something).
1. Something hangs (the page takes too long to load or something).